p2p: standardize outbound full/block relay connection type naming

jonatack opened the pull request:
We've accumulated many different ways of referring to outbound full relay and outbound block relay connection types in the codebase. In places, it is becoming less clear which type of connection is referred to. This PR proposes standardizing the naming to `outbound-{full, block}-relay` with a scripted diff as follows:

```
-BEGIN VERIFY SCRIPT-
s() { git grep -l "$1" src test/functional doc/files.md doc/reduce-*.md | xargs sed -i "s/$1/$2/g"; }

s 'ConnectionType::OUTBOUND '            'ConnectionType::OUTBOUND_FULL_RELAY '
s 'IsFullOutboundConn'                   'IsOutboundFullRelayConn'
s 'GetExtraFullOutboundCount'            'GetExtraOutboundFullRelayCount'
s 'full_outbound_peers'                  'outbound_full_relay_peers'
s 'GetTryNewOutboundPeer'                'GetTryNewOutboundFullRelayPeer'
s 'SetTryNewOutboundPeer'                'SetTryNewOutboundFullRelayPeer'
s 'm_try_another_outbound_peer'          'm_try_another_outbound_full_relay_peer'
s 'outbound (full-relay)'                'outbound-full-relay'
s 'outbound full-relay'                  'outbound-full-relay'
s 'outbound, full-relay'                 'outbound-full-relay'
s 'outbounds'                            'outbound-full-relay connections'
s 'f"outbound: '                         'f"outbound-full-relay: '
s ' full-relay'                          ' outbound-full-relay'
s 'to an extra outbound peer'            'to an extra outbound-full-relay peer'
s 'try another outbound peer'            'try another outbound-full-relay peer'
s '(tx, block, addr) outbound'           '(tx, block, addr)'
s 'we deal with extra outbound peers'    'we deal with extra outbound-full-relay peers'
s 'outbound peers we have in excess'     'outbound-full-relay peers we have in excess'
s 'some outbound connections are not'    'some outbound-full-relay connections are not'
s 'or this is a'                         'or if this is an'

s 'ConnectionType::BLOCK_RELAY'          'ConnectionType::OUTBOUND_BLOCK_RELAY'
s ' BLOCK_RELAY'                         ' OUTBOUND_BLOCK_RELAY'
s 'IsBlockOnlyConn'                      'IsOutboundBlockRelayConn'
s 'GetExtraBlockRelayCount'              'GetExtraOutboundBlockRelayCount'
s 'block_relay_peers'                    'outbound_block_relay_peers'
s 'MAX_BLOCK_RELAY_ONLY_ANCHORS'         'MAX_OUTBOUND_BLOCK_RELAY_ANCHORS'
s 'MAX_BLOCK_RELAY_ONLY_CONNECTIONS'     'MAX_OUTBOUND_BLOCK_RELAY_CONNECTIONS'
s 'EXTRA_BLOCK_RELAY_ONLY_PEER_INTERVAL' 'EXTRA_OUTBOUND_BLOCK_RELAY_PEER_INTERVAL'
s 'm_start_extra_block_relay_peers'      'm_start_extra_outbound_block_relay_peers'
s 'outgoing block-relay-only'            'outbound-block-relay'
s 'outbound block-relay-only'            'outbound-block-relay'
s 'outbound block-relay'                 'outbound-block-relay'
s 'block-relay only outbound'            'outbound-block-relay'
s 'block-relay-only outgoing'            'outbound-block-relay'
s 'block-relay only peers'               'outbound-block-relay peers'
s 'block-relay-only'                     'outbound-block-relay'

s ' a outbound'                          ' an outbound'
-END VERIFY SCRIPT-
```

jonatack committed:
scripted-diff: clean up outbound connection type naming

-BEGIN VERIFY SCRIPT-
s() { git grep -l "$1" src test/functional doc/files.md doc/reduce-*.md | xargs sed -i "s/$1/$2/g"; }

s 'ConnectionType::OUTBOUND '            'ConnectionType::OUTBOUND_FULL_RELAY '
s 'IsFullOutboundConn'                   'IsOutboundFullRelayConn'
s 'GetExtraFullOutboundCount'            'GetExtraOutboundFullRelayCount'
s 'full_outbound_peers'                  'outbound_full_relay_peers'
s 'GetTryNewOutboundPeer'                'GetTryNewOutboundFullRelayPeer'
s 'SetTryNewOutboundPeer'                'SetTryNewOutboundFullRelayPeer'
s 'm_try_another_outbound_peer'          'm_try_another_outbound_full_relay_peer'
s 'outbound (full-relay)'                'outbound-full-relay'
s 'outbound full-relay'                  'outbound-full-relay'
s 'outbound, full-relay'                 'outbound-full-relay'
s 'outbounds'                            'outbound-full-relay connections'
s 'f"outbound: '                         'f"outbound-full-relay: '
s ' full-relay'                          ' outbound-full-relay'
s 'to an extra outbound peer'            'to an extra outbound-full-relay peer'
s 'try another outbound peer'            'try another outbound-full-relay peer'
s '(tx, block, addr) outbound'           '(tx, block, addr)'
s 'we deal with extra outbound peers'    'we deal with extra outbound-full-relay peers'
s 'outbound peers we have in excess'     'outbound-full-relay peers we have in excess'
s 'some outbound connections are not'    'some outbound-full-relay connections are not'
s 'or this is a'                         'or if this is an'

s 'ConnectionType::BLOCK_RELAY'          'ConnectionType::OUTBOUND_BLOCK_RELAY'
s ' BLOCK_RELAY'                         ' OUTBOUND_BLOCK_RELAY'
s 'IsBlockOnlyConn'                      'IsOutboundBlockRelayConn'
s 'GetExtraBlockRelayCount'              'GetExtraOutboundBlockRelayCount'
s 'block_relay_peers'                    'outbound_block_relay_peers'
s 'MAX_BLOCK_RELAY_ONLY_ANCHORS'         'MAX_OUTBOUND_BLOCK_RELAY_ANCHORS'
s 'MAX_BLOCK_RELAY_ONLY_CONNECTIONS'     'MAX_OUTBOUND_BLOCK_RELAY_CONNECTIONS'
s 'EXTRA_BLOCK_RELAY_ONLY_PEER_INTERVAL' 'EXTRA_OUTBOUND_BLOCK_RELAY_PEER_INTERVAL'
s 'm_start_extra_block_relay_peers'      'm_start_extra_outbound_block_relay_peers'
s 'outgoing block-relay-only'            'outbound-block-relay'
s 'outbound block-relay-only'            'outbound-block-relay'
s 'outbound block-relay'                 'outbound-block-relay'
s 'block-relay only outbound'            'outbound-block-relay'
s 'block-relay-only outgoing'            'outbound-block-relay'
s 'block-relay only peers'               'outbound-block-relay peers'
s 'block-relay-only'                     'outbound-block-relay'

s ' a outbound'                          ' an outbound'
-END VERIFY SCRIPT-

jonatack commented:
See the scripted diff in the commit message for details.

practicalswift commented:
Concept ACK: consistent is better than inconsistent

jonatack commented:
rebased

luke-jr commented on src/rpc/net.cpp:
```diff
@@ -31,7 +31,7 @@
 const std::vector<std::string> CONNECTION_TYPE_DOC{
         "outbound-full-relay (default automatic connections)",
-        "block-relay-only (does not relay transactions or addresses)",
+        "outbound-block-relay (does not relay transactions or addresses)",
```
This might be easier to confuse with peers that request blocks only (#20726)

jonatack replied:
Yes, the script can be updated depending on what is decided there.

luke-jr commented:
While doing this, I wonder if it would make sense to also standardise the prefix (eg, "outbound-feeler")

jonatack commented:
> While doing this, I wonder if it would make sense to also standardise the prefix (eg, "outbound-feeler")

On the user-facing side, that may be where it's heading. For instance, in #20764 the -netinfo dir (in/out) column is next to the type column, and the type is expressed as simply full/block/manual/feeler/addr prefixed by the in/out of the dir column (I may propose the same for the GUI peers tab columns). But for now at least, the codebase doesn't have the same variance regarding how a feeler is referred to, like it does with outbound-{full, block}-relay (at least, I don't recall noticing it).

jonatack commented:
rebased

jonatack commented:
Rebased.

> While doing this, I wonder if it would make sense to also standardise the prefix (eg, "outbound-feeler")

Yes, on the user-facing side, that may be where it's heading. For instance, in #20764 the `-netinfo` "dir" (in/out) column is next to the "type" column, and the type is expressed as simply full/block/manual/feeler/addr prefixed by in/out of the dir column (I may propose the same for the GUI peers tab columns). But for now at least, the codebase doesn't have the same variance regarding how a feeler connection is referred to, like it does with `outbound-{full, block}-relay` ones (at least, I don't recall noticing it).

jonatack commented:
Rebased.

jonatack commented:
One concept ACK after 18 days.

Any feedback whether this patch is worth maintaining would be helpful.

ariard reviewed (commented):
ACK 0a6dae5

Thanks for normalizing connection types and +1 for Luke suggestion!

jonatack commented:
Rebased and updated the script per `git range-diff e7eb3712 0a6dae5 de0f38d` due to a non-existing connection type added in `net.h::AddConnection()`, script update:
```diff
+ s 'ConnectionType::OUTBOUND '            'ConnectionType::OUTBOUND_FULL_RELAY '
```

The script may be verified locally after this rebase with:
```
test/lint/commit-script-check.sh e7eb3712..de0f38d5f
```

jonatack commented:
Had to update the script for other changes per `git range-diff e7eb3712 de0f38d bb9f28e`

The script may be verified locally with `test/lint/commit-script-check.sh e7eb371..bb9f28e`

jonatack commented:
Rebased following the merge of #20828. The script may be verified locally with
```
test/lint/commit-script-check.sh 29d2aeb..2645217
```

jonatack commented:
6th rebase. The scripted diff may be verified locally with:
```
test/lint/commit-script-check.sh e9c037b..c3e3a16
```

> ACK 0a6dae5
> 
> Thanks for normalizing connection types and +1 for Luke suggestion!

Thanks @ariard for the review! I'd be happy to do @luke-jr's suggestion in a follow-up. This patch requires frequent rebasing, so if anyone is against it or would like it to be modified (e.g. to some other naming), please LMK (I won't bite :dog2:) to economize rebasing it into the void of the abyss :milky_way:

ariard commented:
Code Review ACK c3e3a16

practicalswift commented:
Lack of rebase is blocking my re-review of this PR :)

jonatack commented:
Thanks everyone for the reviews and ACKs but this commit continually needs rebasing and the issue and diff have become too large without separating this into smaller changes. Up for grabs or maybe will re-open as smaller changes but I'm not sure there is enough support to fix/clean this up.
